File: algorithms/sorting/selection sort/selectionSort.js
const fs = require("fs");
const values = fs
  .readFileSync("values.txt", "utf8")
  .split(",")
  .map((n) => Number(n));

function selectionSort(values) {
  let smallestIndex;
  for (let i = 0; i < values.length; i++) {
    smallestIndex = i;
    for (let j = i + 1; j < values.length; j++) {
      if (values[j] < values[smallestIndex]) {
        smallestIndex = j;
      }
      const temp = values[smallestIndex];
      values[smallestIndex] = values[i];
      values[i] = temp;
    }
  }
  return values;
}

const sortedList = selectionSort(values);
console.log(sortedList);

File: algorithms/sorting/selection sort/selection_sort.py
data = open("values.txt", "r")
values = [int(v) for v in data.read().split(",")]

def selection_sort(values):
    for i in range(len(values)):
        lowest = i
        for j in range(i+1, len(values)):
            if values[j] < values[lowest]:
                lowest = j
        values[i], values[lowest] = values[lowest], values[i]
    print(values)

selection_sort(values)